### PR TITLE
Add vidio.com

### PR DESCRIPTION
**Pull request by MinePlayersPE**
Add guides from vidio.com, for Indonesian channels. Note that there are guides for international channels, but I'm not sure if I should add them too.
Also please, please, please test/check the code first because I literally have no idea what I'm doing

### CODE REVIEW

**Comment by freearhey** [comment]
Thanks for your help! I'll check everything later and be sure to let you know the result.

File: sites/vidio.com.channels.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<site site="vidio.com">
+  <channels>
+    <channel site_id="204" xmltv_id="SCTV.id">SCTV</channel>
+    <channel site_id="205" xmltv_id="Indosiar.id">Indosiar</channel>
+    <channel site_id="665" xmltv_id="RCTI.id">RCTI</channel>
+    <channel site_id="206" xmltv_id="OChannel.id">O Channel</channel>
+    <channel site_id="874" xmltv_id="KompasTV.id">Kompas TV</channel>
+    <channel site_id="6165" xmltv_id="KompasTV.id">BeritaSatu</channel>
+    <channel site_id="777" xmltv_id="BeritaSatu.id">METRO TV</channel>
```
Comment (freearhey) [inline sites/vidio.com.channels.xml:10]: BeritaSatu and METRO TV have the wrong `xmltv_id`

File: sites/vidio.com.config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    items.forEach(item => {
+      const title = (item.querySelector('div.b-livestreaming-daily-schedule__item-content-title') || { textContent: '' }).textContent
+      const time = (item.querySelector('b-livestreaming-daily-schedule__item-content-caption') || { textContent: '' }).textContent
+      const it_date = currdate.add(JSON.parse(item.getAttribute('data-ahoy-props') || '{}').schedule_day, 'day')
+
+      if (title && time) {
+        let start = dayjs(it_date.format('YYYY-MM-DD ').concat(time.substring(0,5)), 'YYYY-MM-DD HH:mm')
+        let stop = dayjs(it_date.format('YYYY-MM-DD ').concat(time.substring(8,13)), 'YYYY-MM-DD HH:mm')
+        if start.diff(stop, 'h') < 0 {
```
Comment (freearhey) [inline sites/vidio.com.config.js:39]: This line returns syntax error. The expression must be in parentheses.

**Review by freearhey — CHANGES_REQUESTED** [review]
At the moment the script only returns an empty program, but in general it looks good. Try running the script on your machine with the command:

```
npx epg-grabber --config=sites/vidio.com.config.js
```

maybe it will help figure out exactly what the problem is.

File: sites/vidio.com.config.js (hunk 2 above)
Comment (MinePlayersPE) [reply sites/vidio.com.config.js:39]: done

Comment (MinePlayersPE) [reply sites/vidio.com.config.js]: done

**Comment by MinePlayersPE** [comment]
Revamped, since old method might not have correctly gotten the time. It seems to work now

File: sites/vidio.com.config.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+dayjs.extend(utc)
+dayjs.extend(customParseFormat)
+dayjs.extend(timezone)
+dayjs.tz.setDefault('Asia/Jakarta')
+
+module.exports = {
+  lang: 'id',
+  site: 'vidio.com',
+  channels: 'vidio.com.channels.xml',
+  output: '.gh-pages/guides/vidio.com.channels.xml',
```
Comment (freearhey) [inline sites/vidio.com.config.js:17]: Should be `.gh-pages/guides/vidio.com.guide.xml`

**Review by freearhey — CHANGES_REQUESTED** [review]
Apart from this small error, everything else is loaded perfectly. But I noticed that the time of the programs does not correspond to what is listed on the site. 

I personally usually use the GSE Smart IPTV to check the schedule, but you can try to load it into any other app and see what should be on the air right now.

**Comment by MinePlayersPE** [comment]
That's weird, when generating the EPG locally and then using the GSE Smart IPTV app the programme times seem to line up with the site. Maybe timezones is the issue?

**Comment by freearhey** [comment]
I think you're right, the problem is the time zone.

For example, "Keajaiban Cinta" on SCTV is supposed to start at `22:00` local time

image

 which is `15:00` UTC, but the guide says 19:00

image

**Comment by MinePlayersPE** [comment]
Well... Like I said, I have barely any idea what I'm doing, but shouldn't me setting the default timezone on the start be enough?

**Comment by freearhey** [comment]
For the guide to work the same for everyone, the `parse()` function must return all dates in UTC.

**Comment by MinePlayersPE** [comment]
Would this solve the issue?

**Comment by freearhey** [comment]
No, it just causes a syntax error.

**Comment by MinePlayersPE** [comment]
Sorry about that, I haven't had enough time to test the code 🙏
I retested it now and it seems to correctly set UTC time, plus the time overflow detection seems to be properly working now

**Comment by freearhey** [comment]
The same result as before:

image

The difference between time on the website and time in the guide must be -7 hours.

**Comment by MinePlayersPE** [comment]
image
Weird, it worked perfectly for me. Hopefully this finally works?

**Comment by freearhey** [comment]
I rewrote your code a bit and the time seems to be correct now. Please check it for yourself and if everything is ok I will add it to the list.

As for the other channels, you can add them all, especially if the program guide for them is also in Indonesian.

**Review by freearhey — APPROVED** [review]

**Comment by MinePlayersPE** [comment]
Sorry for the long wait, there were a few unexpected things this week.
But it seems to work fine on my end too, so LGTM

**Review by freearhey — APPROVED** [review]

**Comment by freearhey** [comment]
Thanks again!

**Pull request merged**